nilearn/plotting/img_plotting.py

```diff
@@ -265,53 +266,46 @@ def _load_anat(anat_img=MNI152TEMPLATE, dim=False, black_bg='auto'):
+        return anat_img, black_bg, vmin, vmax
+
+    if anat_img is MNI152TEMPLATE:
+        anat_img.load()
+        # We special-case the 'canonical anat', as we don't need
+        # to do a few transforms to it.
+        vmin = 0
+        vmax = anat_img.vmax
+        if black_bg == 'auto':
+            black_bg = False
+    else:
+        anat_img = _utils.check_niimg_3d(anat_img)
+        if dim or black_bg == 'auto':
+            # We need to inspect the values of the image
+            data = anat_img.get_data()
+            vmin = data.min()
+            vmax = data.max()
+        if black_bg == 'auto':
+            # Guess if the background is rather black or light based on
+            # the values of voxels near the border
+            background = np.median(get_border_data(data, 2))
+            if background > .5 * (vmin + vmax):
                 black_bg = False
-        else:
-            anat_img = _utils.check_niimg_3d(anat_img)
-            if dim or black_bg == 'auto':
-                # We need to inspect the values of the image
-                data = anat_img.get_data()
-                vmin = data.min()
-                vmax = data.max()
-            if black_bg == 'auto':
-                # Guess if the background is rather black or light based on
-                # the values of voxels near the border
-                border_size = 2
-                border_data = np.concatenate([
-                        data[:border_size, :, :].ravel(),
-                        data[-border_size:, :, :].ravel(),
-                        data[:, :border_size, :].ravel(),
-                        data[:, -border_size:, :].ravel(),
-                        data[:, :, :border_size].ravel(),
-                        data[:, :, -border_size:].ravel(),
-                    ])
-                background = np.median(border_data)
-                if background > .5 * (vmin + vmax):
-                    black_bg = False
-                else:
-                    black_bg = True
-        if dim:
-            vmean = .5 * (vmin + vmax)
-            ptp = .5 * (vmax - vmin)
-            if black_bg:
-                if not isinstance(dim, numbers.Number):
-                    dim = .8
-                vmax = vmean + (1 + dim) * ptp
             else:
-                if not isinstance(dim, numbers.Number):
-                    dim = .6
-                vmin = vmean - (1 + dim) * ptp
-    if black_bg == 'auto':
-        # No anatomy given: no need to turn black_bg on
-        black_bg = False
+                black_bg = True
+    if dim:
+        vmean = .5 * (vmin + vmax)
+        ptp = .5 * (vmax - vmin)
+        if black_bg:
+            if not isinstance(dim, numbers.Number):
+                dim = .8
+            vmax = vmean + (1 + dim) * ptp
+        else:
+            if not isinstance(dim, numbers.Number):
+                dim = .6
+            vmin = vmean - (1 + dim) * ptp
     return anat_img, black_bg, vmin, vmax
 
```

Comment:
This part is slightly tricky, and will probably not fit every use case. On the other hand, explaining the logic of this code is harder than reading it.